src/system.jl

```diff
@@ -53,10 +94,13 @@ Base.@kwdef struct Generator
     ramp_up::Float64
     "Rate at which the generator can decrease generation (pu/minute)"
     ramp_down::Float64
-    "Symbol describing the technology of the generator"
-    technology::Symbol
+    "Type describing the technology of the generator"
+    technology::Technology
 end
-const Generators = Dictionary{UnitCode, Generator}
+# Handle string conversion
+Generator(unit_code::AbstractString, args...) = Generator(InlineString31(unit_code), args...)
```

Comment:
Oh I see!  The only concern I can immediately think of with switching to strings completely is that we index a lot of variables, constraints etc. in the JuMP model by generator unit code.  There could be a noticeable hit to performance if we stop using `Int`s where we have the option.

Reply:
The data structures we use require a lookup rather than a direct index into an array so the difference may not be very noticeable
```julia
julia> x = KeyedArray(1:100, i=101:200);

julia> y = KeyedArray(1:100, i=string.(101:200));

julia> @time y(i="104")
  0.000010 seconds (1 allocation: 16 bytes)
4

julia> @time x(i=104)
  0.000009 seconds (1 allocation: 16 bytes)
4
```